flink-streaming-java/src/test/java/org/apache/flink/streaming/api/graph/StreamingJobGraphGeneratorTest.java

```diff
@@ -805,10 +814,19 @@ public void testSlotSharingOnAllVerticesInSameSlotSharingGroupByDefaultDisabled(
 		final List<JobVertex> verticesSorted = jobGraph.getVerticesSortedTopologicallyFromSources();
 		assertEquals(4, verticesSorted.size());
 
-		final JobVertex source1Vertex = verticesSorted.get(0);
-		final JobVertex source2Vertex = verticesSorted.get(1);
-		final JobVertex map1Vertex = verticesSorted.get(2);
-		final JobVertex map2Vertex = verticesSorted.get(3);
+		JobVertex source1Vertex = null, source2Vertex = null, map1Vertex = null, map2Vertex= null;
+		for (int i = 0; i < 4; i++)
```

Comment:
we could extract this matching block to a common method to avoid duplication. It can return a list of JobVertices in the expected order.

Reply:
Thanks for your comments! I have updated the fix.